Sources/VLCEditController.swift

```diff
@@ -94,29 +96,25 @@ private extension VLCEditController {
 extension VLCEditController: VLCEditToolbarDelegate {
 
     func editToolbarDidAddToPlaylist(_ edittoolbar: VLCEditToolbar) {
-        if let model = model as? PlaylistModel {
-            let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
-                placeHolder: "NEW_PLAYLIST")
-
-            presentTextFieldAlert(with: alertInfo, completionHandler: {
-                text -> Void in
-                    model.create(name: text)
-                })
-
-        } else if let model = model as? VideoModel {
-            let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
-                                               placeHolder: "NEW_PLAYLIST")
-
-            presentTextFieldAlert(with: alertInfo, completionHandler: {
-                [selectedCellIndexPaths, model] text -> Void in
-                let playlist = model.medialibrary.createPlaylist(with: text)
-                for indexPath in selectedCellIndexPaths {
-                    if let media = model.anyfiles[indexPath.row] as? VLCMLMedia {
-                        playlist.appendMedia(withIdentifier: media.identifier())
-                    }
+        //Todo: replace with Viewcontroller that shows existing Playlists
+        let alertInfo = TextFieldAlertInfo(alertTitle: NSLocalizedString("PLAYLISTS", comment: ""),
+                                           alertDescription: NSLocalizedString("PLAYLIST_DESCRIPTION", comment: ""),
+                                           placeHolder: NSLocalizedString("PLAYLIST_PLACEHOLDER", comment:""))
+
+        presentTextFieldAlert(with: alertInfo, completionHandler: {
+            [weak self, selectedCellIndexPaths, model] text -> Void in
+            guard let playlist = self?.services.medialibraryManager.createPlaylist(with: text) else {
```

Comment:
If we want to `weak self`, there is no real need to capture `selectedCellIndexPaths` and `model`.

In this case, if I'm not wrong, since the lifetime of the alert is coupled with the VC, we could just capture `medialibraryManager`.

Reply:
true I can change that